### ISSUE TITLE
Compilation failure on master with gcc 12

### ISSUE THREAD

**Issue by liamwhite** [merryhime/dynarmic#720]
Does not affect use as a submodule; issue is due to -Werror, which is only enabled when dynarmic is the project being built.

```
In member function ‘bool Xbyak::Address::is64bitDisp() const’,
    inlined from ‘void Xbyak::CodeGenerator::opModM(const Xbyak::Address&, const Xbyak::Reg&, int, int, int, int)’ at /home/liam/source/dynarmic/build/externals/xbyak/include/xbyak/xbyak.h:1892:23,
    inlined from ‘void Xbyak::CodeGenerator::opGen(const Xbyak::Operand&, const Xbyak::Operand&, int, int, bool (*)(const Xbyak::Operand&, const Xbyak::Operand&), int, int)’ at /home/liam/source/dynarmic/build/externals/xbyak/include/xbyak/xbyak.h:2006:10,
    inlined from ‘void Xbyak::CodeGenerator::ucomisd(const Xbyak::Xmm&, const Xbyak::Operand&)’ at /home/liam/source/dynarmic/build/externals/xbyak/include/xbyak/xbyak_mnemonic.h:822:56,
    inlined from ‘Dynarmic::Backend::X64::EmitFPMulX<64>(BlockOfCode&, EmitContext&, Dynarmic::IR::Inst*)::<lambda()>::<lambda(auto:55 ...)> [with auto:55 = {Xbyak::Xmm, Xbyak::Xmm}]’ at /home/liam/source/dynarmic/src/dynarmic/backend/x64/emit_x64_floating_point.cpp:757:9,
    inlined from ‘Dynarmic::Backend::X64::EmitFPMulX<64>(BlockOfCode&, EmitContext&, Dynarmic::IR::Inst*)::<lambda()>’ at /home/liam/source/dynarmic/src/dynarmic/backend/x64/emit_x64_floating_point.cpp:757:22,
    inlined from ‘constexpr _Res std::__invoke_impl(__invoke_other, _Fn&&, _Args&& ...) [with _Res = void; _Fn = Dynarmic::Backend::X64::EmitFPMulX<64>(BlockOfCode&, EmitContext&, Dynarmic::IR::Inst*)::<lambda()>&; _Args = {}]’ at /usr/include/c++/12/bits/invoke.h:61:36,
    inlined from ‘constexpr std::enable_if_t<is_invocable_r_v<_Res, _Callable, _Args ...>, _Res> std::__invoke_r(_Callable&&, _Args&& ...) [with _Res = void; _Callable = Dynarmic::Backend::X64::EmitFPMulX<64>(BlockOfCode&, EmitContext&, Dynarmic::IR::Inst*)::<lambda()>&; _Args = {}]’ at /usr/include/c++/12/bits/invoke.h:111:28,
    inlined from ‘static _Res std::_Function_handler<_Res(_ArgTypes ...), _Functor>::_M_invoke(const std::_Any_data&, _ArgTypes&& ...) [with _Res = void; _Functor = Dynarmic::Backend::X64::EmitFPMulX<64>(BlockOfCode&, EmitContext&, Dynarmic::IR::Inst*)::<lambda()>; _ArgTypes = {}]’ at /usr/include/c++/12/bits/std_function.h:290:30:
/home/liam/source/dynarmic/build/externals/xbyak/include/xbyak/xbyak.h:1261:43: warning: array subscript ‘const Xbyak::Address[0]’ is partly outside array bounds of ‘Xbyak::Xmm [1]’ [-Warray-bounds]
 1261 |         bool is64bitDisp() const { return mode_ == M_64bitDisp; } // for moffset
      |                                           ^~~~~
/home/liam/source/dynarmic/src/dynarmic/backend/x64/emit_x64_floating_point.cpp: In static member function ‘static _Res std::_Function_handler<_Res(_ArgTypes ...), _Functor>::_M_invoke(const std::_Any_data&, _ArgTypes&& ...) [with _Res = void; _Functor = Dynarmic::Backend::X64::EmitFPMulX<64>(BlockOfCode&, EmitContext&, Dynarmic::IR::Inst*)::<lambda()>; _ArgTypes = {}]’:
/home/liam/source/dynarmic/src/dynarmic/backend/x64/emit_x64_floating_point.cpp:753:37: note: object ‘args#1’ of size 8
  753 |     ctx.deferred_emits.emplace_back([=, &code] {
      |                                     ^
```

Seems to be mostly bogus.

**Comment by merryhime** [comment]
Known false positive with `-Warray-bounds` within xbyak. <https://github.com/herumi/xbyak/commit/9c8fb81db8d71bb5a1490801bdcc72479b8c52cb#diff-eb93b08b18e882b7b5d442c67d918558f188b0c244efd8ed1e6f50bb586c5378R198>
Can update to 6.68 or later.

**Comment by merryhime** [comment]
Resolved in 916d7cf9bdbb58c7bef26d071ede27e1f99389b2.
